examples/keygen/create_primary.c

```diff
@@ -0,0 +1,248 @@
+/* create_primary.c
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* Tool and example for creating, storing and loading keys using TPM2.0 */
+
+#include <wolftpm/tpm2_wrap.h>
+
+#include <examples/keygen/keygen.h>
+#include <examples/tpm_io.h>
+#include <examples/tpm_test.h>
+#include <examples/tpm_test_keys.h>
+
+#include <stdio.h>
+#include <stdlib.h> /* atoi */
+
+#ifndef WOLFTPM2_NO_WRAPPER
+
+/******************************************************************************/
+/* --- BEGIN TPM Create Primary Key Example -- */
+/******************************************************************************/
+static void usage(void)
+{
+    printf("Expected usage:\n");
+    printf("./examples/keygen/create_primary [-ecc/-rsa] [-oh/eh/ph/] [-unique=] [-auth=] [-aes/xor] [-store=]\n");
+    printf("Primary Key Type:\n");
+    printf("\t-rsa: Use RSA for asymmetric key generation (DEFAULT)\n");
+    printf("\t-ecc: Use ECC for asymmetric key generation \n");
+    printf("Hierarchy:\n");
+    printf("\t-oh: Create keys under the Owner Hierarchy (DEFAULT)\n");
+    printf("\t-eh: Create keys under the Endorsement Hierarchy\n");
+    printf("\t-ph: Create keys under the Platform Hierarchy\n");
+    printf("Unique Template:\n");
+    printf("\t-unique=[value]\n");
+    printf("\t\tOptional unique value for the KDF of the create\n");
+    printf("Authentication:\n");
+    printf("\t-auth=[value]\n");
+    printf("\t\tOptional authentication string for primary\n");
+    printf("Parameter Encryption:\n");
+    printf("\t-aes/xor: Use Parameter Encryption\n");
+    printf("NV Storage:\n");
+    printf("\t-store=[handle]\n");
+    printf("\t\tPersistent primary key handle range: 0x81000000 - 0x810FFFF\n");
+    printf("\t\tUse leading 0x for hex\n");
+
+    printf("Example usage:\n");
+    printf("\t* Create SRK used by wolfTPM:\n");
+    printf("\t\tcreate_primary -rsa -oh -auth=ThisIsMyStorageKeyAuth -store=0x81000200\n");
+}
+
+int TPM2_CreatePrimaryKey_Example(void* userCtx, int argc, char *argv[])
+{
+    int rc;
+    WOLFTPM2_DEV dev;
+    WOLFTPM2_KEY primary;
+    TPMT_PUBLIC publicTemplate;
+    TPMI_ALG_PUBLIC alg = TPM_ALG_RSA; /* default, see usage() for options */
```

Comment:
Comment not needed, but if you want to keep it, add the same thing to `hierarchy` below for consistency.